### PR TITLE
feat(pci-object-storage): download object implementation

### PR DESCRIPTION
**Pull request by seven-amid**
ense          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

**Review by kqesar — APPROVED** [review]

File: packages/manager/apps/pci-object-storage/public/translations/objects/Messages_de_DE.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "pci_projects_project_storages_blocks_add_object_label": "Objektcontainer erstellen",
   "pci_projects_project_storages_containers_my_containers_label": "Meine Container",
   "pci_projects_project_storages_containers_s3_users_label": "S3-Nutzer",
+  "pci_projects_project_storages_containers_container_object_error_download": "Beim Download des Objekts ist ein Fehler aufgetreten: {{ message }}.",
```
Comment (tibs245) [inline packages/manager/apps/pci-object-storage/public/translations/objects/Messages_de_DE.json:5]: The translation is provided from CMT ?

File: packages/manager/apps/pci-object-storage/src/pages/objects/container/object/download/Download.page.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                },
+              )
+            }
+          </Translation>,
+          true,
+        ),
+      );
+  };
+
+  return <></>;
```
Comment (tibs245) [inline packages/manager/apps/pci-object-storage/src/pages/objects/container/object/download/Download.page.tsx:62]: I not agree this conception. Can you speak it together ?

File: packages/manager/apps/pci-object-storage/src/api/data/objects.ts [hunk 3]
```diff
@@ -65,3 +66,38 @@ export const addUser = async (
...
+  const { data } = await v6.post(
+    `/cloud/project/${projectId}/storage/${storageId}/publicUrl`,
+    {
+      expirationDate,
+      objectName: object?.name,
+    },
+  );
+
+  return data;
+};
```
Comment (tibs245) [inline packages/manager/apps/pci-object-storage/src/api/data/objects.ts:103]: Use hook "use..." and maybe use mutation

File: packages/manager/apps/pci-object-storage/package.json [hunk 4]
```diff
@@ -40,7 +40,8 @@
     "react-i18next": "^14.0.5",
     "react-router-dom": "^6.24.1",
     "react-use": "^17.5.0",
-    "zustand": "^4.5.5"
+    "zustand": "^4.5.5",
```
Comment (tibs245) [inline packages/manager/apps/pci-object-storage/package.json:43]: Why this is added but not used ?

**Review by tibs245 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**